bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/americanexpress/AmericanExpress.java

```diff
@@ -0,0 +1,192 @@
+import com.liato.bankdroid.banking.exceptions.BankException;
+import com.liato.bankdroid.banking.exceptions.LoginException;
+import com.liato.bankdroid.legacy.R;
+import com.liato.bankdroid.provider.IAccountTypes;
+import com.liato.bankdroid.provider.IBankTypes;
+
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpStatus;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.protocol.HTTP;
+import org.joda.time.format.DateTimeFormat;
+
+import android.content.Context;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+
+import eu.nullbyte.android.urllib.CertificateReader;
+import eu.nullbyte.android.urllib.Urllib;
+
+public class AmericanExpress extends Bank {
+
+    private static final String TAG = "AmericanExpress";
+
+    private static final String NAME = "American Express";
+
+    private static final String NAME_SHORT = "americanexpress";
+
+    private static final String URL = "https://www.americanexpress.com/home/se/home_c.shtml";
+
+    private static final int BANKTYPE_ID = IBankTypes.AMERICANEXPRESS;
+
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    private LoginResponse loginResponse;
+
+    public AmericanExpress(Context context) {
+        super(context, R.drawable.logo_americanexpress);
+        super.TAG = TAG;
+        super.NAME = NAME;
+        super.NAME_SHORT = NAME_SHORT;
+        super.BANKTYPE_ID = BANKTYPE_ID;
+        super.URL = URL;
+        super.WEB_VIEW_ENABLED = false;
+    }
+
+    public AmericanExpress(String username, String password, Context context) throws BankException,
+            LoginException, BankChoiceException, IOException {
+        this(context);
+        this.update(username, password);
+    }
+
+    @Override
+    protected LoginPackage preLogin() throws BankException, IOException {
+        urlopen = new Urllib(context, CertificateReader.getCertificates(context,
+                R.raw.cert_americanexpress, R.raw.cert_americanexpress_global));
+        urlopen.setAllowCircularRedirects(true);
+        urlopen.setContentCharset(HTTP.ISO_8859_1);
+        urlopen.addHeader("Face", "sv_SE");
+        return new LoginPackage(urlopen, null, null,
+                "https://global.americanexpress.com/myca/intl/moblclient/emea/svc/v1/loginSummary.do");
+    }
+
+    @Override
+    public Urllib login() throws LoginException, BankException, IOException {
+        LoginPackage lp = preLogin();
+
+        HttpResponse response = urlopen.openAsHttpResponse(lp.getLoginTarget(),
+                new StringEntity(objectAsJson(new LoginRequest(getUsername(), getPassword())),
+                        HTTP.UTF_8), true);
+
+        loginResponse = MAPPER.reader()
+                .withType(LoginResponse.class)
+                .readValue(response.getEntity().getContent());
+        if(loginResponse.getLogonData().getStatus() != 0) {
```

Comment:
You can get Android Studio to highlight these things for you as you type.

Go to Settings / Editor / Inspections / Java / Constant conditions & exceptions.

Enable the inspection, and also make sure to check "Suggest @Nullable annotations ...".

Now you should get to see this as you type, rather than as-you-do-PRs :). Yay, short feedback cycles!